src/containers/ProfilePage/index.jsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import { Container, Row } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import UserDetail from "../../components/UserDetail";
+import UserEmail from "../../components/UserEmail";
+import UserPassword from "../../components/UserPassword";
+import defaultProfileImage from "../../images/defaultProfileImage.png";
+import "./index.scss";
+
+function getArrowClass(selectedElm, currentElm) {
+  return selectedElm === currentElm ? "arrow down" : "arrow right";
+}
+
+function checkCollapseClass(selectedElm, currentElm) {
+  if (selectedElm === currentElm) return "";
+
+  return `collapsed`;
+}
+
+export default function ProfilePage({ submit }) {
+  const [selectedElement, setSelectedElement] = React.useState("userDetail");
+  const { register, handleSubmit, errors } = useForm();
+
+  const onSave = (data) => {
+    // TODO: save data in the database
+    submit(data);
+  };
+
+  return (
+    <Container fluid="md" className="profileContainer">
+      <form onSubmit={handleSubmit(onSave)}>
+        <Row className="profilePortrait">
+          <h2>Profile</h2>
+          <div className="userPortrait">
+            <img src={defaultProfileImage} alt="User portrait" />
+          </div>
+        </Row>
+        <Row className="userDetail">
+          <button
+            className="sectionTitle"
+            onClick={() => setSelectedElement("userDetail")}
+            type="button"
+          >
+            <i className={getArrowClass(selectedElement, "userDetail")} />
+            User Details
+          </button>
+          <div
+            className={`
+            sectionContent 
+            ${checkCollapseClass(selectedElement, "userDetail")}
+          `}
+          >
+            <UserDetail register={register} validErrors={errors} />
```

Comment:
is this supposed to be an input or what? how long is this component going to be, if its not complicated it can just be inlined rather than being made a separate element

Reply:
it's not an input, a component that hold the inputs related to user detail, it's too long so I decided to split it